client_types.go

```diff
@@ -33,51 +33,35 @@ type Metadata struct {
 	Release       string            `json:"release"`
 }
 
-//nolint:maligned
-type jsonRequest struct {
-	DirectDomains         *[]string `json:"direct_domains"`
-	DomainNames           []string  `json:"domain_names"`
-	ExtraInfo             *string   `json:"extra_info"`
-	FastFailRegexps       *[]string `json:"fast_fail_regexps"`
-	Metadata              Metadata  `json:"metadata"`
-	NoProxyCaching        bool      `json:"no_proxy_caching"`
-	NoSSLBumpDomains      *[]string `json:"no_ssl_bump_domains"`
-	Protocol              *string   `json:"protocol"`
-	SharedTunnel          bool      `json:"shared_tunnel"`
-	SquidConfig           *string   `json:"squid_config"`
-	SSHPort               int       `json:"ssh_port"`
-	TLSPassthroughDomains *[]string `json:"tls_passthrough_domains"`
-	TLSResignDomains      *[]string `json:"tls_resign_domains"`
-	TunnelIdentifier      *string   `json:"tunnel_identifier"`
-	TunnelPool            bool      `json:"tunnel_pool"`
-	VMVersion             *string   `json:"vm_version"`
-}
-
 // Request for a new tunnel.
-//
-//nolint:maligned
 type Request struct {
-	DomainNames      []string
-	TunnelIdentifier string
-
-	DirectDomains         []string
-	FastFailRegexps       []string
-	KGPPort               int
-	NoProxyCaching        bool
-	NoSSLBumpDomains      []string
-	Protocol              string
-	SharedTunnel          bool
-	TLSPassthroughDomains []string
-	TLSResignDomains      []string
-	TunnelPool            bool
-	VMVersion             string
+	TunnelIdentifier string `json:"tunnel_identifier"`
+	Protocol         string `json:"protocol"`
 
-	// Metadata contains the request metadata.
-	Metadata Metadata
+	SharedTunnel bool `json:"shared_tunnel"`
+	TunnelPool   bool `json:"tunnel_pool"`
 
-	// Extra info. This is a string (which contains a JSON dict) to enable
-	// optional features and flags.
-	ExtraInfo string
+	TunnelDomains         []string `json:"tunnel_domains,omitempty"`
+	DirectDomains         []string `json:"direct_domains,omitempty"`
```

Comment:
The `direct_domains` value is expected to be a list of domains (and not regexps) for SC-4. We have quite a mess rn but we'd clean it up.

OK (messy but we fix that after the client is updated):

`deny_domains` - regexps for sc5
`fast_fail_regexps` - regexps for sc4
`tls_passthrough_domains` - regexps for sc5
`no_ssl_bump_domains` - domains for sc4
`tunnel_domains` - regexps for sc5
`domain_names` - domains for sc4

Not OK:
`direct_domains` - could be domain or regexp.

We would need to make the REST API doesn't validate `direct_domains`  for SC5. It's doable and it's, arguably better than adding a new `direct_regexps` flags or smth like that.

Reply:
Agreed. We need to do some requests marking / versioning, adding `direct_regexps` just increases the entropy.

The `experimental` flag will not work because it's passed from server to client.

I think we can use the new `--metadata` flag for that, the `--metadata` values will be put to [Metadata.Extra field](https://github.com/saucelabs/tunnelrest-go/blob/2ae2aaeb47b5102d5cdb96a86b4c646ea10a9312/client_types.go#L26) so we would have `--metadata api=v5alpha`

WDYT @amckenzie132

Reply:
Or simpler just use the Protocol value?

Reply:
> I think we can use the new --metadata flag for that

sounds good. The Protocol value is not enough during the transition. Some `--metadata` value should do it